### PR TITLE
Add a spell checker

### PR DESCRIPTION
**Pull request by harupy**
<details><summary>&#x1F6E0 DevTools &#x1F6E0</summary>
<p>

[Open in GitHub Codespaces](https://codespaces.new/harupy/mlflow/pull/13868?quickstart=1)

#### Install mlflow from this PR

```
pip install git+https://github.com/mlflow/mlflow.git@refs/pull/13868/merge
```

#### Checkout with GitHub CLI

```
gh pr checkout 13868
```

</p>
</details>

### Related Issues/PRs

 #xxx

### What changes are proposed in this pull request?

Add a spell checker to prevent typos.

### How is this PR tested?

- [x] Existing unit/integration tests
- [ ] New unit/integration tests
- [ ] Manual tests

### Does this PR require documentation update?

- [x] No. You can skip the rest of this section.
- [ ] Yes. I've updated:
  - [ ] Examples
  - [ ] API references
  - [ ] Instructions

#### Is this a user-facing change?

- [ ] No. You can skip the rest of this section.
- [ ] Yes. Give a description of this change to be included in the release notes for MLflow users.

#### What component(s), interfaces, languages, and integrations does this PR affect?

Components

- [ ] `area/artifacts`: Artifact stores and artifact logging
- [ ] `area/build`: Build and test infrastructure for MLflow
- [ ] `area/deployments`: MLflow Deployments client APIs, server, and third-party Deployments integrations
- [ ] `area/docs`: MLflow documentation pages
- [ ] `area/examples`: Example code
- [ ] `area/model-registry`: Model Registry service, APIs, and the fluent client calls for Model Registry
- [ ] `area/models`: MLmodel format, model serialization/deserialization, flavors
- [ ] `area/recipes`: Recipes, Recipe APIs, Recipe configs, Recipe Templates
- [ ] `area/projects`: MLproject format, project running backends
- [ ] `area/scoring`: MLflow Model server, model deployment tools, Spark UDFs
- [ ] `area/server-infra`: MLflow Tracking server backend
- [ ] `area/tracking`: Tracking Service, tracking client APIs, autologging

Interface

- [ ] `area/uiux`: Front-end, user experience, plotting, JavaScript, JavaScript dev server
- [ ] `area/docker`: Docker use across MLflow's components, such as MLflow Projects and MLflow Models
- [ ] `area/sqlalchemy`: Use of SQLAlchemy in the Tracking Service or Model Registry
- [ ] `area/windows`: Windows support

Language

- [ ] `language/r`: R APIs and clients
- [ ] `language/java`: Java APIs and clients
- [ ] `language/new`: Proposals for new client languages

Integrations

- [ ] `integrations/azure`: Azure and Azure ML integrations
- [ ] `integrations/sagemaker`: SageMaker integrations
- [ ] `integrations/databricks`: Databricks integrations

<a name="release-note-category"></a>

#### How should the PR be classified in the release notes? Choose one:

- [x] `rn/none` - No description will be included. The PR will be mentioned only by the PR number in the "Small Bugfixes and Documentation Updates" section
- [ ] `rn/breaking-change` - The PR will be mentioned in the "Breaking Changes" section
- [ ] `rn/feature` - A new user-facing feature worth mentioning in the release notes
- [ ] `rn/bug-fix` - A user-facing bug fix worth mentioning in the release notes
- [ ] `rn/documentation` - A user-facing documentation change worth mentioning in the release notes

#### Should this PR be included in the next patch release?

`Yes` should be selected for bug fixes, documentation updates, and other small changes. `No` should be selected for new features and larger changes. If you're unsure about the release classification of this PR, leave this unchecked to let the maintainers decide.

<details>
<summary>What is a minor/patch release?</summary>

- Minor release: a release that increments the second part of the version number (e.g., 1.2.0 -> 1.3.0).
  Bug fixes, doc updates and new features usually go into minor releases.
- Patch release: a release that increments the third part of the version number (e.g., 1.2.0 -> 1.2.1).
  Bug fixes and doc updates usually go into patch releases.

</details>

- [ ] Yes (this PR will be cherry-picked and included in the next patch release)
- [x] No (this PR will be included in the next minor release)

### CODE REVIEW

File: tests/evaluate/logging/test_utils.py [hunk 1]
```diff
@@ -161,7 +161,7 @@ def test_evaluations_to_dataframes_basic():
     assert evaluations_df["inputs_id"].iloc[0] == "inputs1"
     assert evaluations_df["inputs"].iloc[0] == {"feature1": 1.0, "feature2": 2.0}
 
-    # Check that the othe
+    # Check that the other
```
Comment (harupy) [inline tests/evaluate/logging/test_utils.py:164]:
```suggestion
```

File: tests/tracing/export/test_inference_table_exporter.py [hunk 2]
```diff
@@ -25,7 +25,7 @@ def test_export():
         span_id=1,
         parent_id=None,
         start_time=0,
-        end_time=1_000_000,  # 1 milisecond
+        end_time=1_000_000,  # 1 millisecond
```
Comment (harupy) [inline tests/tracing/export/test_inference_table_exporter.py:28]:
```suggestion
        end_time=1_000_000,  # 1 milisecond
```

test

File: mlflow/pytorch/_lightning_autolog.py [hunk 3]
```diff
@@ -46,7 +46,7 @@
 
 def _get_optimizer_name(optimizer):
     """
-    In pytorch-lightining 1.1.0, `LightningOptimizer` was introduced:
+    In pytorch-lighting 1.1.0, `LightningOptimizer` was introduced:
```
Comment (BenWilson2) [inline mlflow/pytorch/_lightning_autolog.py:49]:
```suggestion
    In pytorch-lightning 1.1.0, `LightningOptimizer` was introduced:
```

**Review by BenWilson2 — APPROVED** [review]
Looking great so far! Thanks for adding this @harupy :D

**Review by WeichenXu123 — APPROVED** [review]
LGTM!

**Comment by nojaf** [comment]
Was this added to the devcontainer as well? It does not seem to be present for me there.

**Pull request merged**